src/main/java/org/kcctl/command/DeleteOffsetsCommand.java

```diff
@@ -0,0 +1,89 @@
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.eclipse.microprofile.rest.client.RestClientBuilder;
+import org.kcctl.completion.ConnectorNameCompletions;
+import org.kcctl.service.ConnectorOffsets;
+import org.kcctl.service.KafkaConnectApi;
+import org.kcctl.util.ConfigurationContext;
+import org.kcctl.util.Version;
+import picocli.CommandLine;
+
+import javax.inject.Inject;
+import java.util.Set;
+import java.util.concurrent.Callable;
+
+/**
+ * Deletes the committed offsets for a connector.
+ *
+ * @see <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-875%3A+First-class+offsets+support+in+Kafka+Connect">KIP-875</a>
+ */
+@CommandLine.Command(name = "offsets", description = "Delete committed connector offsets")
+public class DeleteOffsetsCommand implements Callable<Integer> {
+
+    @CommandLine.Mixin
+    HelpMixin help;
+
+    private final Version requiredVersionForDeletingOffsets = new Version(3, 6);
+    private final ObjectMapper mapper = new ObjectMapper();
+    private final ConfigurationContext context;
+
+    @CommandLine.Spec
+    CommandLine.Model.CommandSpec spec;
+
+    @CommandLine.Parameters(paramLabel = "NAME", description = "Name(s) of the connector(s) (e.g. 'my-connector')", completionCandidates = ConnectorNameCompletions.class)
+    Set<String> names = Set.of();
+
+    @Inject
+    public DeleteOffsetsCommand(ConfigurationContext context) {
+        this.context = context;
+    }
+
+    // Hack : Picocli currently require an empty constructor to generate the completion file
+    public DeleteOffsetsCommand() {
+        context = new ConfigurationContext();
+    }
+
+    @Override
+    public Integer call() throws JsonProcessingException {
+        KafkaConnectApi kafkaConnectApi = RestClientBuilder.newBuilder()
+                .baseUri(context.getCurrentContext().getCluster())
+                .build(KafkaConnectApi.class);
+
+        Version currentVersion = new Version(kafkaConnectApi.getWorkerInfo().version());
+
+        if (!currentVersion.greaterOrEquals(requiredVersionForDeletingOffsets)) {
+            spec.commandLine().getErr().println(String.format("Deleting connector offsets requires at least Kafka Connect %s. Current version: %s",
+                    requiredVersionForDeletingOffsets, currentVersion));
+            return 1;
+        }
+
+        for (String connector : names) {
+            ConnectorOffsets offsets = kafkaConnectApi.deleteConnectorOffsets(connector);
+            // Display with JSON; the contents aren't very readable as raw text, and this output format
+            // plays nicely with the endpoint to patch connector offsets (you can use the output here as the
+            // body for requests to that endpoint)
+            String offsetsJson = mapper.writerWithDefaultPrettyPrinter().writeValueAsString(offsets);
+            spec.commandLine().getOut().println(offsetsJson);
```

Comment:
Is this correct? I thought the responses for altering/resetting offsets were different from the ones for reading them. The [KIP](https://cwiki.apache.org/confluence/display/KAFKA/KIP-875%3A+First-class+offsets+support+in+Kafka+Connect#KIP875:FirstclassoffsetssupportinKafkaConnect-Altering/resettingoffsets(response)) indicates that it should be something like this:

```json
{
  "message": "The offsets for this connector have been reset successfully"
}
```

or this:
```json
{
  "message": "The framework-managed offsets for this connector have been reset successfully. However, if this connector manages offsets externally, they will need to be manually reset in the system that the connector uses."
}
```

Reply:
You are right. I'll make the changes that are needed. Good catch :wink: